src/Http/Routing/src/Matching/LinearSearchJumpTable.cs

```diff
@@ -34,14 +34,7 @@ public override int GetDestination(string path, PathSegment segment)
             for (var i = 0; i < entries.Length; i++)
             {
                 var text = entries[i].text;
-                if (segment.Length == text.Length &&
-                    string.Compare(
-                        path,
-                        segment.Start,
-                        text,
-                        0,
-                        segment.Length,
-                        StringComparison.OrdinalIgnoreCase) == 0)
+                if (path.Length == text.Length && path.Equals(text, StringComparison.OrdinalIgnoreCase))
```

Comment:
```suggestion
                if (path.Equals(text, StringComparison.OrdinalIgnoreCase))
```
The length-check is done [in equals as well](https://github.com/dotnet/runtime/blob/bd6cbe3642f51d70839912a6a666e5de747ad581/src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.Globalization.cs#L81-L87), so omit one.